src/services/query.ts

```diff
@@ -116,3 +128,16 @@ const alphabeticallyByLabels = (a: Term, b: Term) => {
   const sortLabelB = prefLabelB + altLabelB;
   return sortLabelA.localeCompare(sortLabelB);
 };
+
+function guardTimeout<T>(
+  promise: Promise<T>,
+  timeoutMs: number,
+  timeoutError: TimeoutError
+): Promise<T> {
+  return Promise.race([
+    promise,
+    new Promise(resolve =>
+      setTimeout(resolve.bind(null, timeoutError), timeoutMs)
```

Comment:
Is it useful to log a (informational?) message in case of a timeout, to aid debugging?

Reply:
As the error message returned to clients is quite clear, we decided not to add logging for now:

```json
"result": {
  "__typename": "TimeoutError",
  "message": "Source timed out after 1ms"
}
```